CompileFilter crashes while logging error if the filter is missing brackets

Example:
```
package main

import "github.com/go-ldap/ldap"

func main() {
	ldap.CompileFilter("((cn=)")
}
```

When running the example, it fails to parse the filter and crashes while logging the error message:
```./[foo@localhost test]# ./test
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
panic(0x54ef60, 0xc82000a0e0)
	/usr/lib/golang/src/runtime/panic.go:481 +0x3e6
github.com/go-ldap/ldap.CompileFilter(0x57d790, 0x6, 0xc820014810, 0x0, 0x0)
	/root/src/github.com/go-ldap/ldap/filter.go:86 +0x41e
main.main()
	/root/test/test.go:6 +0x2c
```

The problem is with this line in ldap/filter.go, pos gets invalid value:

`return nil, NewError(ErrorFilterCompile, errors.New("ldap: finished compiling filter with extra at end: "+fmt.Sprint(filter[pos:])))`

--- a/filter.go
+++ b/filter.go
@@ -82,7 +82,10 @@ func CompileFilter(filter string) (*ber.Packet, error) {
 	if err != nil {
 		return nil, err
 	}
-	if pos != len(filter) {
+	switch {
+	case pos > len(filter):
+		return nil, NewError(ErrorFilterCompile, errors.New("ldap: unexpected end of filter"))
+	case pos < len(filter):
 		return nil, NewError(ErrorFilterCompile, errors.New("ldap: finished compiling filter with extra at end: "+fmt.Sprint(filter[pos:])))
 	}
 	return packet, nil

--- a/filter_test.go
+++ b/filter_test.go
@@ -132,6 +132,12 @@ var testFilters = []compileTest{
 		expectedErr:    "unexpected end of filter",
 	},
 	compileTest{
+		filterStr:      `((cn=)`,
+		expectedFilter: ``,
+		expectedType:   0,
+		expectedErr:    "unexpected end of filter",
+	},
+	compileTest{
 		filterStr:      `(&(objectclass=inetorgperson)(cn=中文))`,
 		expectedFilter: `(&(objectclass=inetorgperson)(cn=\e4\b8\ad\e6\96\87))`,
 		expectedType:   0,

